packages/relay/src/lib/services/cacheService/cacheService.ts

```diff
@@ -216,10 +218,13 @@ export class CacheService {
         this.logger.error(
           `${requestIdPrefix} Error occurred while setting the cache to Redis. Fallback to internal cache. Error is: ${redisError.fullError}`,
         );
+
+        // fallback to internal cache in case of Redis error
+        this.cacheMethodsCounter.labels(callingMethod, CacheService.cacheTypes.LRU, CacheService.methods.SET).inc(1);
+        await this.internalCache.set(key, value, callingMethod, ttl, requestIdPrefix);
       }
     } else {
       this.cacheMethodsCounter.labels(callingMethod, CacheService.cacheTypes.LRU, CacheService.methods.SET).inc(1);
-
       await this.internalCache.set(key, value, callingMethod, ttl, requestIdPrefix);
     }
   }
```

Comment:
**nit:** We can avoid duplicating the fallback logic inside the `catch` and `else` blocks by adding a `return` statement if the main logic finishes successfully:

```typescript
if (this.isSharedCacheEnabled) {
  try {
     this.cacheMethodsCounter.labels(...).inc(1);
     await this.sharedCache.set(...);
     return;
  } catch {
    this.logger.error('...');
  }
}
// fallback
this.cacheMethodsCounter.labels(...).inc(1);
await this.internalCache.set(...);
```